### PR TITLE
Replace standard burner-wallet balances and add passports support

### LINKED ISSUE #2: Replace standard burner-wallet balances with CO2, Goellars and Trees
## Scope

- Currently the burner-wallet features three crypto currencies DAI, ETH, PDAI
- In the "Planet A" game, we'd like to replace the above currencies with three new "assets":
  - Goellars: The in game currency that functions as a "virtual dollar"
  - CO2: An asset that describes the amount a player decided to externalize their cost
  - Plants: An asset that describes the amount a player decided to offset their external cost ("Planting trees reduces CO2")
- We're not planning to run the game at the Ethereum meetup on mainnet! Instead we're aiming for the LeapDAO test network.
- Goellars and CO2 are not exchangeable (so there's no need for an "Exchange.js" integration")
- Goellars, CO2 are already registered as LeapDAO test network tokens. Their colors can be found here: https://testnet.leapdao.org/registerToken
- As neither Goellars nor CO2 are intended to be exited from Plasma, there's also no need to have any ETH in the wallet. Hence the ETH balance can be hidden too

Examples where e.g. PDAI is used as the default currency but should be replaced with Goellars:

Screen Shot 2019-07-16 at 14 05 19
Screen Shot 2019-07-16 at 14 05 08
Screen Shot 2019-07-16 at 14 05 02

## Deliverables

- Replace DAI, ETH, PDAI balances with Goellars, Plants and CO2 in App.js
- Make sure that *only* Goellars are transferrable in the UI between users. Plants and CO2 are not supposed to be transferrable
- Adjust the balance of Goellars in other places too e.g. SendToAddress.js and others

## Roles
bounty gardener: @TimDaub / 10%
bounty worker: name / 75%
bounty reviewer: name / 15%

### PR DESCRIPTION
**Pull request by MaxStalker**
Fixes #2

### CODE REVIEW

File: src/components/GoellarsBalance.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    return <span key={type}>{type === "currency" ? "₲" : value}</span>;
+  });
+};
+
+const GoellarsBalance = props => {
+  const { balance } = props;
+  const locale = getStoredValue("i18nextLng") || "en.en";
+  const formatter = new Intl.NumberFormat(locale, {
+    style: "currency",
+    currency: "PYG", // We use Paraguayan guaraní to display ₲ next to a value
```
Comment (vrde) [inline src/components/GoellarsBalance.js:38]: image

File: src/config.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ const configs = [
     SIDECHAIN: {
       NAME: "Leap Testnet",
       RPC: "https://testnet-node.leapdao.org",
-      DAI_ADDRESS: "0xD2D0F8a6ADfF16C2098101087f9548465EC96C98",
+      DAI_ADDRESS: "0x1f89Fb2199220a350287B162B9D0A330A2D2eFAD",
+      DAI_SYMBOL: "₲",
+      DAI_SHORTNAME: "GÖLR",
```
Comment (vrde) [inline src/config.js:13]: should we add those values also for the `toco2` domain in the config?

File: src/services/plasma.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    .slice(2) // strip 0x
+    .slice(start * 2, end * 2)}`;
+};
+export const extractData = passport => {
+  const rawData = passport.output.data;
+
+  const nameHex = sliceHex(rawData, 0, 20);
+  const imageHex = sliceHex(rawData, 20, 24);
+  const emittedHex = sliceHex(rawData, 24, 28);
+  const lockedHex = sliceHex(rawData, 28, 32);
```
Comment (vrde) [inline src/services/plasma.js]: It's the other way around, first locked, then emitted:

```
+------------+------------+------------+-------------+
|20 bytes    | 4 bytes    | 4 bytes    | 4 bytes     |
|name str    | picId      | CO2 locked | CO2 emitted |
+------------+------------+------------+-------------+
```

File: src/services/plasma.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+export const getCitizenship = async (plasma, account) => {
+  const citizenship = getStoredValue("citizenship", account);
+  if (!citizenship) {
+    fetchPassportData(plasma, account);
+  } else {
+    return citizenship;
+  }
+};
+
+export const fetchAlPassports = async (plasma, account) => {
```
Comment (vrde) [inline src/services/plasma.js]: typo: missing an `l` → `fetchAllPassports`

**Review by vrde — CHANGES_REQUESTED** [review]

File: src/config.js (hunk 2 above)
Comment (MaxStalker) [reply src/config.js:13]: That's a good question :)
Can we move this one out to XSS (or maybe even unpaid) issue ?

File: src/services/plasma.js (hunk 3 above)
Comment (MaxStalker) [reply src/services/plasma.js]: Fixed

File: src/services/plasma.js (hunk 4 above)
Comment (MaxStalker) [reply src/services/plasma.js]: Nice spot! Fixed that one ;)

File: src/components/GoellarsBalance.js (hunk 1 above)
Comment (MaxStalker) [reply src/components/GoellarsBalance.js:38]: In the end this one didn't work and I was needed to render currency with symbol "by hand" :\

File: src/config.js (hunk 2 above)
Comment (MaxStalker) [reply src/config.js:13]: There are some more values like "BRIDGE_ADDRESS" and "UNISWAP" addresses, which is incorrect, though unused in PlanetA.

Comment (MaxStalker) [reply src/config.js:13]: I've updated other values though, but would be cool if they are working correctly on that domain.

File: src/components/Passports/PassportView.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      <IconClose onClick={close} />
+      <Citizen name={name} country={country} />
+      <Flex
+        pb={3}
+        mb={3}
+        borderBottom={"1px solid #DFDFDF"}
+        width="100%"
+        justifyContent="space-around"
+      >
+        <Param label={"CO2 Produced"} value={emitted} color="emitted" />
```
Comment (vrde) [inline src/components/Passports/PassportView.js]:
```suggestion
        <Param label={"CO₂ Produced"} value={emitted} color="emitted" />
```

Comment (vrde) [inline src/components/Passports/PassportView.js]:
```suggestion
        <Param label={"CO₂ Produced"} value={emitted} color="emitted" />
```

File: src/components/Passports/PassportView.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        <Param label={"CO2 Locked"} value={locked} color="locked" />
```
Comment (vrde) [inline src/components/Passports/PassportView.js]:
```suggestion
        <Param label={"CO₂ Locked"} value={locked} color="locked" />
```

File: src/components/Passports/styles.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+}))`
+  border-radius: 4px 8px 8px 4px;
+  box-shadow: inset -1px 0 rgba(255, 255, 255, 0.2),
+    0 1px 3px rgba(0, 0, 0, 0.2);
+  position: relative;
+  border: 2px solid #333;
+
+  @supports (background-blend-mode: color-burn) {
+    background-blend-mode: color-burn;
+    background-image: url("https://www.toptal.com/designers/subtlepatterns/patterns/papyrus.png");
```
Comment (vrde) [inline src/components/Passports/styles.js]: I would avoid hotlinking the background. Can we make it as a local asset by downloading it?

Remember to add the credits, something along those lines would work:

> Background made by Olivia Harmon, https://www.toptal.com/designers/subtlepatterns/papyrus-pattern/

File: src/components/Passports/index.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        : null;
+    }
+    this.state = initialState;
+
+    // Bind methods to class
+    this.selectPassport = this.selectPassport.bind(this);
+    this.closePassport = this.closePassport.bind(this);
+  }
+
+  componentWillReceiveProps({ list }) {
```
Comment (vrde) [inline src/components/Passports/index.js]: I'm not familiar with `componentWillReceiveProps`, but I read that this method (and others) [will be deprecated](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops). Do you think we can have another approach here?

File: src/services/plasma.js (hunk 3 above)
Comment (vrde) [reply src/services/plasma.js]: I still see the old version :thinking: 

image

**Review by vrde — CHANGES_REQUESTED** [review]

File: src/components/Passports/styles.js (hunk 7 above)
Comment (MaxStalker) [reply src/components/Passports/styles.js]: Good point :)

File: src/services/plasma.js (hunk 3 above)
Comment (MaxStalker) [reply src/services/plasma.js]: Forgot to actually push...
Will also add formatting to that number

File: src/components/Passports/index.js (hunk 8 above)
Comment (MaxStalker) [reply src/components/Passports/index.js]: Good call :)

File: src/components/Passports/PassportView.js [hunk 9]
```diff
@@ -3,13 +3,17 @@ import { Flex, Text } from "rimble-ui";
 import { IconClose } from "./styles";
 
 const Param = ({ label, value, color }) => {
+  const locale = localStorage.getItem('i18nextLng')
+  const formatter = new Intl.NumberFormat(locale, {
+    minimumFractionDigits: 2,
```
Comment (vrde) [inline src/components/Passports/PassportView.js:8]: Not sure we need this, but we can fix it later.

**Review by vrde — APPROVED** [review]
Looks great! Not sure we need the decimal digits, but we can fix it later.

**Pull request merged**